pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -215,6 +215,39 @@ func chooseRandomNode(windowsVMs map[types.WindowsVM]bool) types.WindowsVM {
 	return nil
 }
 
+// deleteWindowsVM terminates the VM with the given id through the cloud provider
+func (r *ReconcileWindowsMachineConfig) terminateWindowsVM(id string) error {
+	// Delete the Windows VM from cloud provider
+	if err := r.cloudProvider.DestroyWindowsVM(id); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("could not delete VM with ID %s", id))
+	}
+	return nil
+}
+
+// delete removes the given vm from the list of VMs and terminates the underlying VM
+func (r *ReconcileWindowsMachineConfig) delete(vm types.WindowsVM) error {
+	// VM is missing credentials, this can occur if there was a failure initially creating it. We can consider the
+	// actual VM terminated as there is nothing we can do with it.
+	if vm.GetCredentials() == nil || len(vm.GetCredentials().GetInstanceId()) == 0 {
+		delete(r.windowsVMs, vm)
+		return nil
+	}
+
+	// Terminate the instance via its instance id
+	id := vm.GetCredentials().GetInstanceId()
+	log.V(1).Info("deleting the Windows VM", "ID", id)
+	if err := r.terminateWindowsVM(id); err != nil {
```

Comment:
Won't calling `r.cloudProvider.DestroyWindowsVM(id)` have the same effect here instead of the helper?